packages/o-spreadsheet-engine/src/plugins/ui_stateful/filter_evaluation.ts

```diff
@@ -139,6 +139,11 @@ export class FilterEvaluationPlugin extends UIPlugin {
     const id = this.getters.getFilterId({ sheetId, col, row });
     if (!id) return;
     if (!this.filterValues[sheetId]) this.filterValues[sheetId] = {};
+    if (value.filterType === "values") {
+      value.hiddenValues = value.hiddenValues.map(toTrimmedLowerCase);
+    } else if (value.filterType === "criterion") {
+      value.values = value.values.map(toTrimmedLowerCase);
```

Comment:
criterion values can be formulas (eg. `=A1`), in which case lowercasing the formula is useless. You should lowercase the evaluated formula value instead (`evaluatedCriterionValues`)